crates/ty_python_semantic/src/semantic_model.rs

```diff
@@ -69,12 +69,12 @@ impl<'db> SemanticModel<'db> {
             return vec![];
         };
         let ty = Type::module_literal(self.db, self.file, &module);
+        let builtin = module.is_known(KnownModule::Builtins);
         crate::types::all_members(self.db, ty)
             .into_iter()
-            .map(|name| Completion {
-                name,
-                builtin: module.is_known(KnownModule::Builtins),
-            })
+            // Filter out private members from `builtins`
+            .filter(|name| !builtin || !name.starts_with('_'))
+            .map(|name| Completion { name, builtin })
```

Comment:
We could also filter _after_ constructing the `Completion`, or filter _elsewhere_ in the file. I'm not familiar enough to gauge the trade-offs here, but this spot seems reasonable.

Reply:
Regarding `_` vs `__`, see https://github.com/astral-sh/ruff/pull/19102/files#r2180951944 and the OP.